src/analyze/analyze.c

```diff
@@ -2397,6 +2402,13 @@ static int parse_argv(int argc, char *argv[]) {
                                 return r;
                         break;
 
+                case ARG_THRESHOLD:
+                        r = safe_atou(optarg, &arg_threshold);
+                        if (r < 0 || arg_threshold > 100)
+                                return log_error_errno(r, "Failed to parse threshold: %s", optarg);
```

Comment:
if arg_threshold is > 100 then r will be zero and you return success here. i.e. change first arg to log_error_errno() to `r < 0 ? r : SYNTHETIC_ERRNO(EINVAL)`